pilot/pkg/security/trustdomain/bundle.go

```diff
@@ -84,9 +85,9 @@ func (t Bundle) ReplaceTrustDomainAliases(principals []string) []string {
 // replaceTrustDomains replace the given principal's trust domain with the trust domains from the
 // trustDomains list and return the new principals.
 func (t Bundle) replaceTrustDomains(principal, trustDomainFromPrincipal string) []string {
-	principalsForAliases := []string{}
+	principalsForAliases := sets.Set{}
```

Comment:
In most cases there will be only a few items in the `principalsForAliases` and `t.TrustDomains` list, this likely won't be any faster especially you have to do an extra `SortedList()` in the end.